common/selectors/wallet.js

```diff
@@ -39,3 +40,25 @@ export function getTokenBalances(state: State): TokenBalance[] {
     custom: t.custom
   }));
 }
+
+export function getTxFromTransactionsBySignedTx(
+  state: State,
+  signedTx: string
+): ?BroadcastStatusTransaction {
+  const transactions = state.wallet.transactions;
+  const matchingTxs = transactions.filter(function(obj) {
+    return obj.signedTx === signedTx;
+  });
+  return matchingTxs ? matchingTxs[0] : null;
+}
+
+// TODO type signedTx?
+export function getTxFromBroadcastStatusTransactions(
+  transactions: Array<BroadcastStatusTransaction>,
+  signedTx: string
+): ?BroadcastStatusTransaction {
+  const matchingTxs = transactions.filter(function(obj) {
+    return obj.signedTx === signedTx;
+  });
+  return matchingTxs ? matchingTxs[0] : null;
```

Comment:
`Array.prototype.find` is a little more succinct.

Reply:
agreed.